src/model/balance.js

```diff
@@ -60,13 +63,27 @@ class Balance extends EventEmitter {
     return this.add(this._convert(amount).negated().toString())
   }
 
+  isValidOutgoing (amountString) {
+    let amount = this._convert(amountString)
+    return this._getNumber().then((balance) => {
+      let inMax = balance.add(amount).lte(this._max)
+      let inWarn = balance.add(amount).lte(this._warnMax)
+      let positive = amount.gte(this._convert('0'))
+      if (!inWarn) {
+        this.emit('over', balance)
+      }
+      return Promise.resolve(inMax && positive)
+    })
+  }
+
   isValidIncoming (amountString) {
     let amount = this._convert(amountString)
     return this._getNumber().then((balance) => {
       let inLimit = balance.sub(amount).gte(this._limit.negated())
+      let inWarn = balance.sub(amount).gte(this._warnLimit.negated())
       let positive = amount.gte(this._convert('0'))
-      if (!inLimit) {
-        this.emit('settlement', balance)
+      if (!inWarn) {
+        this.emit('under', balance)
```

Comment:
A function called `isValid*` probably shouldn't have side effects. You could have a separate method `maybeTriggerSettlement`.